app/view/settingView.js

```diff
@@ -75,8 +71,9 @@ export default class SettingView extends React.Component {
             <div style={{flexBasis: 33}}/>
             <div style={{flexBasis: 128}}>
               <Select
-                options={options}
-                onChange={()=>{}}
+                options={lang.languages}
+                onChange={(val)=>this.props.setSourceLanguage(val)}
```

Comment:
Couldnt we just code it like 
`{this.props.setSourceLanguage}`
If it does not work, please let me know. Im not sure right now